### ISSUE TITLE
Record patterns

### ISSUE THREAD

**Issue by asr** [ndmitchell/hlint#187]
Running hlint on

```haskell
data A = A Int Int Int

foo :: A -> Bool
foo (A _ _ _) = True
```

it suggests

```bash
Warning: Use record patterns
Found:
  A _ _ _
Why not:
  A{}
```

but running hlint on

```haskell
data B = B Int Int

bar :: B -> Bool
bar (B _ _) = True
```

there are not suggestion. Is this behaviour on purpose?

**Comment by mpickering** [comment]
It does. You just need to have three or more wildcards for it to trigger the hint.

For example,

```
data B = B Int Int

bar :: B -> Bool
bar (B _ _ _) = True
```

**Comment by asr** [comment]
But your example doesn't compile.

**Comment by mpickering** [comment]
I meant,

```
data B = B Int Int Int

bar :: B -> Bool
bar (B _ _ _) = True
```

**Comment by ndmitchell** [comment]
The hint is valid on any number of fields, but A usually looks cleaner than A{}, while (A _ _ _ _ _ _) is clearly worse. I picked 3 fields as being the first point it seems unambiguously better, but I do usually use this approach even at one field, as it gives better extensibility if I change the constructor.

**Comment by asr** [comment]
Thanks for the explanation.
